### PR TITLE
restructured the upgrade methods in new file

### PR DESCRIPTION
**Pull request by pragamj**
Signed-off-by: pragam <pragam.jain@seagate.com>

# Problem Statement
- Restructured the Upgrade code.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
- https://eos-jenkins.colo.seagate.com/job/QA/job/K8s_Cortx_Continuous_Deployment/869/console
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)
[collections_log.txt](https://github.com/Seagate/cortx-test/files/8831836/collections_log.txt)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: libs/prov/prov_k8s_cortx_upgrade.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+from commons.helpers.pods_helper import LogicalNode
+from libs.prov.prov_k8s_cortx_deploy import ProvDeployK8sCortxLib
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ProvUpgradeK8sCortxLib:
+    """
+    This class contains utility methods for all the operations related
+    to k8s based Cortx Deployment .
```
Comment (pranav-sahasrabudhe) [inline libs/prov/prov_k8s_cortx_upgrade.py:36]: Need to update from `Deployment` to `Upgrade`

**Review by pranav-sahasrabudhe — CHANGES_REQUESTED** [review]
Please check review and do the neeful. Thanks!

File: libs/prov/prov_k8s_cortx_upgrade.py (hunk 1 above)
Comment (pragamj) [reply libs/prov/prov_k8s_cortx_upgrade.py:36]: done

File: config/prov/prov_test.yaml [hunk 2]
```diff
@@ -149,6 +149,10 @@ k8s_cortx_deploy:
   git_k8_repo: "https://github.com/Seagate/cortx-k8s.git"
   git_k8_repo_file: "https://raw.githubusercontent.com/Seagate/cortx-k8s/$tag/k8_cortx_cloud/$file"
   k8s_dir: "/root/deploy-scripts/k8_cortx_cloud/"
+  prov_branch: "cortx-test"
+  upg_type_cold: "cold"
+  upg_type_roll: "rolling"
+  gran_type: "all"
```
[the remark this answers is not in the document]
Comment (pragamj) [reply config/prov/prov_test.yaml]: done

**Review by pranav-sahasrabudhe — APPROVED** [review]

File: scripts/cicd_k8s_cortx_deploy/create_db_entry.py [hunk 3]
```diff
@@ -89,6 +98,7 @@ def create_db_entry(hosts, cfg, admin_user, admin_pswd, nodes_cnt, s3_engine, po
     json_data["s3_engine"] = int(s3_engine)
     json_data["product_type"] = "k8s"
     json_data["setup_in_useby"] = "CICD_Deployment"
+    json_data["lb"] = ext_ip
```
Comment (priyankaborawake) [inline scripts/cicd_k8s_cortx_deploy/create_db_entry.py:101]: Port no also needs to be appended to ip

**Review by priyankaborawake — APPROVED** [review]

**Review by ketanarlulkar — DISMISSED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**